cpp/tests/hashing/xxhash64_test.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/iterator.cuh>
+#include <cudf/fixed_point/fixed_point.hpp>
+#include <cudf/hashing.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+using NumericTypesNoBools =
+  cudf::test::Concat<cudf::test::IntegralTypesNotBool, cudf::test::FloatingPointTypes>;
+
+template <typename T>
+class HashXX64TestTyped : public cudf::test::BaseFixture {};
+
+TYPED_TEST_SUITE(HashXX64TestTyped, NumericTypesNoBools);
+
+TYPED_TEST(HashXX64TestTyped, TestAllNumeric)
+{
+  using T   = TypeParam;
+  auto col1 = cudf::test::fixed_width_column_wrapper<T, int32_t>{
+    {-1, -1, 0, 2, 22, 0, 11, 12, 116, 32, 0, 42, 7, 62, 1, -22, 0, 0},
+    {1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0}};
+  auto col2 = cudf::test::fixed_width_column_wrapper<T, int32_t>{
+    {-1, -1, 0, 2, 22, 1, 11, 12, 116, 32, 0, 42, 7, 62, 1, -22, 1, -22},
+    {1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0}};
+
+  auto output1 = cudf::hashing::xxhash64(cudf::table_view({col1}));
+  auto output2 = cudf::hashing::xxhash64(cudf::table_view({col2}));
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output1->view(), output2->view());
+
+  output1 = cudf::hashing::xxhash64(cudf::table_view({col1}), 7);
+  output2 = cudf::hashing::xxhash64(cudf::table_view({col2}), 7);
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(output1->view(), output2->view());
+}
+
+class HashXX64Test : public cudf::test::BaseFixture {};
+
+TEST_F(HashXX64Test, TestInteger)
+{
+  auto col1 =
+    cudf::test::fixed_width_column_wrapper<int32_t>{{-127,
+                                                     -70000,
+                                                     0,
+                                                     200000,
+                                                     128,
+                                                     std::numeric_limits<int32_t>::max(),
+                                                     std::numeric_limits<int32_t>::min(),
+                                                     std::numeric_limits<int32_t>::lowest()}};
+
+  auto const output = cudf::hashing::xxhash64(cudf::table_view({col1}));
+
+  // these were generated using the CPU compiled version of the cuco xxhash64 source
```

Comment:
Can you independently verify these results with a 3rd party implementation of xxhash, maybe the Python library https://pypi.org/project/xxhash/? Just want to be sure we cover our bases since there are always weird bugs in hashing functions...

Reply:
> Can you independently verify these results with a 3rd party implementation of xxhash, maybe the Python library https://pypi.org/project/xxhash/? Just want to be sure we cover our bases since there are always weird bugs in hashing functions...

I was able to verify the results match the results from this Python library